mailer.http.class.php

```diff
@@ -43,50 +56,64 @@ function sparkpostSend()
         $this->edebug('Response received');
 
         return $this->handle_response($result);
-
-
     }
 
+    /**
+     * Build the request body to be sent to the SparkPost API.
+     */
     protected function get_request_body()
     {
         $tracking_enabled = !!$this->options['enable_tracking'];
         $sender = $this->get_sender();
-        $body = array(
-            'recipients' => $this->get_recipients(),
-            'content' => array(
-                'from' => $sender,
-                'subject' => $this->Subject,
-                'headers' => $this->get_headers()
-            ),
-            'options' => array(
-                'open_tracking' => $tracking_enabled,
-                'click_tracking' => $tracking_enabled
-            )
+        $replyTo = $this->get_reply_to();
+        $body = array();
+
+        // add recipients
+        $body['recipients'] = $this->get_recipients();
+
+        // enable engagement tracking
+        $body['options'] = array(
+            'open_tracking' => $tracking_enabled,
+            'click_tracking' => $tracking_enabled
         );
 
+        // pass through either stored template or inline content
         if (!empty($this->options['template'])) {
-          $body['content']['template_id'] = $this->options['template'];
```

Comment:
We could do that later - right now I'm just trying to address the immediate customer issues and people are running out of time. Can we create a ticket to do that?